### Description of the data set

In [ ]:
#Importation of Libraries used
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import plotly.express as px

In [ ]:
#Importation of data from a local place
#Data is available on https://www.kaggle.com/radmirzosimov/telecom-users-dataset

data = pd.read_csv("telecom_users.csv",index_col="customerID")
data.head()

In [110]:
#Shape of the data set
data.shape

(5986, 21)

In [111]:
data.isnull().sum()

Unnamed: 0          0
gender              0
SeniorCitizen       0
Partner             0
Dependents          0
tenure              0
PhoneService        0
MultipleLines       0
InternetService     0
OnlineSecurity      0
OnlineBackup        0
DeviceProtection    0
TechSupport         0
StreamingTV         0
StreamingMovies     0
Contract            0
PaperlessBilling    0
PaymentMethod       0
MonthlyCharges      0
TotalCharges        0
Churn               0
dtype: int64

The dataset contains 5986 users described by 21 variables with no missing values. The "Unnamed: 0" will be drop for the need of the analysis because we do not have information about it.

In [112]:
#Drop "Unnamed: 0" column
data = data.drop(columns=["Unnamed: 0"])
data.columns.tolist()

['gender',
 'SeniorCitizen',
 'Partner',
 'Dependents',
 'tenure',
 'PhoneService',
 'MultipleLines',
 'InternetService',
 'OnlineSecurity',
 'OnlineBackup',
 'DeviceProtection',
 'TechSupport',
 'StreamingTV',
 'StreamingMovies',
 'Contract',
 'PaperlessBilling',
 'PaymentMethod',
 'MonthlyCharges',
 'TotalCharges',
 'Churn']

In [113]:
data.dtypes

gender               object
SeniorCitizen         int64
Partner              object
Dependents           object
tenure                int64
PhoneService         object
MultipleLines        object
InternetService      object
OnlineSecurity       object
OnlineBackup         object
DeviceProtection     object
TechSupport          object
StreamingTV          object
StreamingMovies      object
Contract             object
PaperlessBilling     object
PaymentMethod        object
MonthlyCharges      float64
TotalCharges         object
Churn                object
dtype: object

TotalCharges must be in a numerical type. So that, we have to convert into a numerci form (float).

In [62]:
data["TotalCharges"] = str(data["TotalCharges"])

In [114]:
Empty_TotalCharges = data.loc[data.TotalCharges == " "]
Empty_TotalCharges.shape

(10, 20)

In [118]:
data[data.tenure == 0]

gender  SeniorCitizen Partner Dependents  tenure PhoneService  \
customerID                                                                  
2775-SEFEE    Male              0      No        Yes       0          Yes   
1371-DWPAZ  Female              0     Yes        Yes       0           No   
3213-VVOLG    Male              0     Yes        Yes       0          Yes   
2923-ARZLG    Male              0     Yes        Yes       0          Yes   
7644-OMVMY    Male              0     Yes        Yes       0          Yes   
5709-LVOEQ  Female              0     Yes        Yes       0          Yes   
3115-CZMZD    Male              0      No        Yes       0          Yes   
2520-SGTTA  Female              0     Yes        Yes       0          Yes   
4472-LVYGI  Female              0     Yes        Yes       0           No   
4367-NUYAO    Male              0     Yes        Yes       0          Yes   

               MultipleLines InternetService       OnlineSecurity  \
customerID                                                          
2775-SEFEE               Yes             DSL                  Yes   
1371-DWPAZ  No phone service             DSL                  Yes   
3213-VVOLG               Yes              No  No internet service   
2923-ARZLG                No              No  No internet service   
7644-OMVMY                No              No  No internet service   
5709-LVOEQ                No             DSL                  Yes   
3115-CZMZD                No              No  No internet service   
2520-SGTTA                No              No  No internet service   
4472-LVYGI  No phone service             DSL                  Yes   
4367-NUYAO               Yes              No  No internet service   

                   OnlineBackup     DeviceProtection          TechSupport  \
customerID                                                                  
2775-SEFEE                  Yes                   No                  Yes   
1371-DWPAZ                  Yes                  Yes                  Yes   
3213-VVOLG  No internet service  No internet service  No internet service   
2923-ARZLG  No internet service  No internet service  No internet service   
7644-OMVMY  No internet service  No internet service  No internet service   
5709-LVOEQ                  Yes                  Yes                   No   
3115-CZMZD  No internet service  No internet service  No internet service   
2520-SGTTA  No internet service  No internet service  No internet service   
4472-LVYGI                   No                  Yes                  Yes   
4367-NUYAO  No internet service  No internet service  No internet service   

                    StreamingTV      StreamingMovies  Contract  \
customerID                                                       
2775-SEFEE                   No                   No  Two year   
1371-DWPAZ                  Yes                   No  Two year   
3213-VVOLG  No internet service  No internet service  Two year   
2923-ARZLG  No internet service  No internet service  One year   
7644-OMVMY  No internet service  No internet service  Two year   
5709-LVOEQ                  Yes                  Yes  Two year   
3115-CZMZD  No internet service  No internet service  Two year   
2520-SGTTA  No internet service  No internet service  Two year   
4472-LVYGI                  Yes                   No  Two year   
4367-NUYAO  No internet service  No internet service  Two year   

           PaperlessBilling              PaymentMethod  MonthlyCharges  \
customerID                                                               
2775-SEFEE              Yes  Bank transfer (automatic)           61.90   
1371-DWPAZ               No    Credit card (automatic)           56.05   
3213-VVOLG               No               Mailed check           25.35   
2923-ARZLG              Yes               Mailed check           19.70   
7644-OMVMY               No               Mailed check           19.85   
5709-LVOEQ            

In [115]:
Empty_TotalCharges

gender  SeniorCitizen Partner Dependents  tenure PhoneService  \
customerID                                                                  
2775-SEFEE    Male              0      No        Yes       0          Yes   
1371-DWPAZ  Female              0     Yes        Yes       0           No   
3213-VVOLG    Male              0     Yes        Yes       0          Yes   
2923-ARZLG    Male              0     Yes        Yes       0          Yes   
7644-OMVMY    Male              0     Yes        Yes       0          Yes   
5709-LVOEQ  Female              0     Yes        Yes       0          Yes   
3115-CZMZD    Male              0      No        Yes       0          Yes   
2520-SGTTA  Female              0     Yes        Yes       0          Yes   
4472-LVYGI  Female              0     Yes        Yes       0           No   
4367-NUYAO    Male              0     Yes        Yes       0          Yes   

               MultipleLines InternetService       OnlineSecurity  \
customerID                                                          
2775-SEFEE               Yes             DSL                  Yes   
1371-DWPAZ  No phone service             DSL                  Yes   
3213-VVOLG               Yes              No  No internet service   
2923-ARZLG                No              No  No internet service   
7644-OMVMY                No              No  No internet service   
5709-LVOEQ                No             DSL                  Yes   
3115-CZMZD                No              No  No internet service   
2520-SGTTA                No              No  No internet service   
4472-LVYGI  No phone service             DSL                  Yes   
4367-NUYAO               Yes              No  No internet service   

                   OnlineBackup     DeviceProtection          TechSupport  \
customerID                                                                  
2775-SEFEE                  Yes                   No                  Yes   
1371-DWPAZ                  Yes                  Yes                  Yes   
3213-VVOLG  No internet service  No internet service  No internet service   
2923-ARZLG  No internet service  No internet service  No internet service   
7644-OMVMY  No internet service  No internet service  No internet service   
5709-LVOEQ                  Yes                  Yes                   No   
3115-CZMZD  No internet service  No internet service  No internet service   
2520-SGTTA  No internet service  No internet service  No internet service   
4472-LVYGI                   No                  Yes                  Yes   
4367-NUYAO  No internet service  No internet service  No internet service   

                    StreamingTV      StreamingMovies  Contract  \
customerID                                                       
2775-SEFEE                   No                   No  Two year   
1371-DWPAZ                  Yes                   No  Two year   
3213-VVOLG  No internet service  No internet service  Two year   
2923-ARZLG  No internet service  No internet service  One year   
7644-OMVMY  No internet service  No internet service  Two year   
5709-LVOEQ                  Yes                  Yes  Two year   
3115-CZMZD  No internet service  No internet service  Two year   
2520-SGTTA  No internet service  No internet service  Two year   
4472-LVYGI                  Yes                   No  Two year   
4367-NUYAO  No internet service  No internet service  Two year   

           PaperlessBilling              PaymentMethod  MonthlyCharges  \
customerID                                                               
2775-SEFEE              Yes  Bank transfer (automatic)           61.90   
1371-DWPAZ               No    Credit card (automatic)           56.05   
3213-VVOLG               No               Mailed check           25.35   
2923-ARZLG              Yes               Mailed check           19.70   
7644-OMVMY               No               Mailed check           19.85   
5709-LVOEQ            

In [116]:
data.loc[data.TotalCharges == " ","TotalCharges"] = 0.0

The fact that we have some empty value in TotalCharges column

In [117]:
data.dtypes

gender               object
SeniorCitizen         int64
Partner              object
Dependents           object
tenure                int64
PhoneService         object
MultipleLines        object
InternetService      object
OnlineSecurity       object
OnlineBackup         object
DeviceProtection     object
TechSupport          object
StreamingTV          object
StreamingMovies      object
Contract             object
PaperlessBilling     object
PaymentMethod        object
MonthlyCharges      float64
TotalCharges         object
Churn                object
dtype: object

In [121]:
test = data.TotalCharges.values
test = test.float

AttributeError: 'numpy.ndarray' object has no attribute 'float'

### Demographics characteristics of users

### Analysis of subscribed services

### Prediction of churn rate using a Logistic regression